libflux/src/flux/formatter/mod.rs

```diff
@@ -1,32 +1,57 @@
 #![allow(missing_docs)]
-use crate::ast::{self, walk::Node};
+use crate::ast::{self, walk::Node, File};
+use crate::parser::parse_string;
 use crate::Error;
 
 use chrono::SecondsFormat;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn format_from_js_file(js_file: JsValue) -> String {
```

Comment:
does &str work?

Reply:
for the return value or the function argument?

Reply:
for the return value see the above comment

Reply:
Looks like this should be a JsString, but no need to clone it
https://github.com/rustwasm/wasm-bindgen/issues/1705